image/image.go

```diff
@@ -0,0 +1,68 @@
+package image
+
+import (
+	"encoding/json"
+
+	"github.com/buildpack/imgutil"
+	"github.com/pkg/errors"
+
+	"github.com/buildpack/pack/style"
+)
+
+type Image interface {
+	Raw() imgutil.Image
```

Comment:
This seems like a code smell. Seems like we should be able to use **embedding** (analogous to delegate pattern in OOP) to effectively adhere to the `imgutil.Image` interface?

For example:
```go
type Image interface {
	imgutil.Image
}

type image struct {
	imgutil.Image
}
```

Ref: https://yourbasic.org/golang/inheritance-object-oriented/

Reply:
Seems like you are using embedding in other places like `stack.Image`. Is there a limitation that I'm not seeing here?

Reply:
I suppose you could, but I wanted to limit how much these wrappers can do at their top-level interface, to keep them simple and focused. `Raw()` just gives a way to do what you need to in a pinch (when the top-level interface doesn't suffice).

My hope is that eventually, `Raw()` will go away completely, once the pertinent manipulations are exposed as appropriate methods on each wrapper type.

Reply:
Removal of `Raw()` is something we might even be able to pull off in this PR

Reply:
Although I agree with the goals of the future I think that the fact that `Raw` is currently there would mean that it would break the API anyway in the future yet if we embed the `imgutil.Image` we get a much cleaner API for how things currently exist.

Reply:
Keep in mind PR #375. I wouldn't think these images would be part of the exported API, at least not initially.